### PR TITLE
Acceptance: Handle numerical version names in version comparison helpers

### PR DESCRIPTION
**Pull request by mandre**
There are two changes here:
- rename the `IsReleasesBelow()` and `IsReleasesAbove()` functions to `IsCurrentBelow()` and `IsCurrentAbove()` respectively.
- fix their behavior to account for numerical release names, after zed, and ensure that
comparing the same version always yields false. Also fixing bugs when comparing to master and add comprehensive tests.

### CODE REVIEW

**Comment by EmilienM** [comment]
I changed the label manually before our workflow doesn't know (yet) that it's acceptance and that API is not maintained here.
I also fixed the failure via https://github.com/gophercloud/gophercloud/pull/2765 .

File: acceptance/clients/conditions_test.go
Comment (EmilienM) [inline acceptance/clients/conditions_test.go:2]: Please put that file into a `testing directory` or this will be skipped by the unittest (in fact coverage) job. See https://github.com/gophercloud/gophercloud/blob/4f6d1451acac9260a6901451e146b2bee4742bb5/script/coverage#L6

**Comment by mandre** [comment]
The [go-apidiff failure](https://github.com/gophercloud/gophercloud/actions/runs/6118191513/job/16606621557?pr=2764) should be fixed with https://github.com/joelanford/go-apidiff/commit/05117854c081149093a1282693d26426dc5b3d0e. Waiting for a new release.

**Comment by EmilienM** [comment]
> The [go-apidiff failure](https://github.com/gophercloud/gophercloud/actions/runs/6118191513/job/16606621557?pr=2764) should be fixed with [joelanford/go-apidiff@0511785](https://github.com/joelanford/go-apidiff/commit/05117854c081149093a1282693d26426dc5b3d0e). Waiting for a new release.

no, you'll need to rebase your PR on top of https://github.com/gophercloud/gophercloud/pull/2773 once merged, and the job will be happy. However the job will set the label to `semver:major` because again we don't make a difference of API chances in the directories, for now.

**Review by EmilienM — APPROVED** [review]

**Pull request merged**
